librarian_server/api/admin.py

```diff
@@ -96,3 +100,50 @@ def add_file(
     session.commit()
 
     return AdminCreateFileResponse(success=True, file_exists=True)
+
+
+@router.post(
+    "/verify_file",
+    response_model=Union[FileVerificationResponse, AdminRequestFailedResponse],
+)
+def verify_file(
+    request: AdminVerifyFileRequest,
+    response: Response,
+    session: Session = Depends(yield_session),
+) -> Union[FileVerificationResponse, AdminRequestFailedResponse]:
+    """
+    Verifies the properties of an existing file in the database and returns newly computed checksums and sizes for all of the instances.
+    If the requested store or file does not exist, or if no instances are found, it returns a 400 Bad Request response.
+    """
+
+    file = session.get(File, request.name)
+    if file is None:
+        response.status_code = status.HTTP_400_BAD_REQUEST
+        return AdminRequestFailedResponse(
+            reason="File not found.",
+            suggested_remedy="Ensure the file exists in the database.",
+        )
+
+    instances = file.instances
+    if not instances:
+        response.status_code = status.HTTP_400_BAD_REQUEST
+        return AdminRequestFailedResponse(
+            reason="File instances not found.",
+            suggested_remedy="Ensure the file has instances in the database.",
+        )
+
+    checksums_and_sizes = []
+    for instance in instances:
+        path = instance.path
+        checksum = get_md5_from_path(path)
+        size = get_size_from_path(path)
+        checksums_and_sizes.append(
+            {
+                "store_id": str(instance.store_id),
+                "checksum": checksum,
+                "size": str(size),
+            }
```

Comment:
You will want to check here if the instance is available, or it will not be there and we will get a shock! We should probably check that we have _at least_ one instance on the librarian?

We frequently 'remove' instances (for example during SneakerNet transfers)

Reply:
What do you mean by Instance? We check the file instance in the above code!!